### PR TITLE
re-enable codecov, and sonarqube on the frontend.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: .github/workflows/app-react.yml [hunk 1]
```diff
@@ -57,26 +57,20 @@ jobs:
...
-          env_vars: javascript
-          # Specify whether or not CI build should fail if Codecov runs into an error during upload
-          fail_ci_if_error: false
+          name: pr-codecov
+          path: pr/
 
       - name: SonarQube Scan
         id: scan
         uses: sonarsource/sonarqube-scan-action@master
-        if: false
```
Comment (devinleighsmith) [inline .github/workflows/app-react.yml:79]: I'm not sure why this was disabled - the quality gate / upload pass on my fork.

File: .github/workflows/api-dotnetcore.yml [hunk 2]
```diff
@@ -104,21 +104,16 @@ jobs:
           cd ..
           head TestResults/coverage.opencover.xml
 
-      - name: Codecov
-        uses: codecov/codecov-action@v3.1.1
-        env:
-          CODECOV_TOKEN: ${{ secrets.CODECOV }}
+      - name: Save PR number and scan results
```
Comment (devinleighsmith) [inline .github/workflows/api-dotnetcore.yml:107]: upload test results for consumption by codecov.

File: .github/workflows/codecov-comment-pr.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        uses: actions/github-script@v3.1.0
+        with:
+          script: |
+            var artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "pr-codecov"
```
Comment (devinleighsmith) [inline .github/workflows/codecov-comment-pr.yml:28]: pull the named artifact.

File: .github/workflows/codecov-comment-pr.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          # User defined upload name. Visible in Codecov UI
+          name: PIMS
+          # Path to coverage file to upload
+          file: ./coverage.opencover.xml
+          # Flag upload to group coverage metrics (e.g. unittests | integration | ui,chrome)
+          flags: unittests
+          env_vars: C#
+          # Specify whether or not CI build should fail if Codecov runs into an error during upload
+          fail_ci_if_error: false
+          override_pr: ${{steps.create-comment.outputs.result}}
```
Comment (devinleighsmith) [inline .github/workflows/codecov-comment-pr.yml:70]: Need to provide the PR, as this workflow does not have direct access to the PR.

File: .github/workflows/codecov-comment-pr.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
+name: Codecov Comment on PR
+
+# read-write repo token
+# access to secrets
+on:
+  workflow_run:
+    workflows: ["API (.NET 8)", "APP (React)"]
```
Comment (devinleighsmith) [inline .github/workflows/codecov-comment-pr.yml:7]: Kicked off by the completion of one of these two.

File: .github/workflows/codecov-comment-pr.yml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            await github.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: issue_number,
+              body: 'See CodeCov Report Here: https://app.codecov.io/github/bcgov/psp/pull/' + issue_number
+            });
+            return issue_number;
+          result-encoding: string
+      - name: Codecov(API)
+        if: ${{ github.event.workflow_run.name == 'API (.NET 8)' }}
```
Comment (devinleighsmith) [inline .github/workflows/codecov-comment-pr.yml:56]: slightly different params for the two different workflows.

**Comment by devinleighsmith** [comment]
Tested this on my fork, doesn't provide the nice visual that codecov used to with this format, can figure that out later.

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**